src/arrayMethodSort.js

```diff
@@ -4,9 +4,27 @@
  * Implement method Sort
  */
 function applyCustomSort() {
-  [].__proto__.sort2 = function(compareFunction) {
-    // write code here
+  const defaultCompareFun = (a, b) => a.toString() > b.toString();
+
+  [].__proto__.sort2 = function(compareFunction = defaultCompareFun) {
+    let counter;
+
+    do {
+      counter = 0;
+
+      for (let i = 0; i < this.length - 1; i++) {
+        const a = this[i];
+        const b = this[i + 1];
+
+        if (compareFunction !== undefined && compareFunction(a, b) > 0) {
```

Comment:
Using default function parameter we dont need to check if function is not undefined
```suggestion
        if (compareFunction(a, b) > 0) {
```